pkg/plugin/plugin.go

```diff
@@ -100,6 +100,24 @@ func (p *plugin) GenerateARM(ctx context.Context, cs *api.OpenShiftManagedCluste
 	return p.armGenerator.Generate(ctx, cs, isUpdate)
 }
 
+func (p *plugin) RecoverEtcdCluster(ctx context.Context, cs *api.OpenShiftManagedCluster, azuretemplate map[string]interface{}, deployFn api.DeployFn) *api.PluginError {
+	p.log.Info("deleting master scaleset")
+	if err := p.clusterUpgrader.DeleteScaleSet(ctx, cs, "ss-master"); err != nil {
```

Comment:
I've got a few worries.
1) what happens if the restored etcd has a different view of what nodes should be around than what is actually there?
2) how can we make very sure we don't accidentally run the restore a second time around?

I'm wondering whether this workflow should look more like the following:
1) delete *all* the scalesets and the hash blob.
2) run a deploy with the flag which includes the restore functionality
3) run an update with the flag unset (should just rotate the masters removing the script addition)

Still need to manually check (and then have an automated test) to cover the differing nodes problem.

@kargakis @mjudeikis thoughts here?

Reply:
I am not sure deleting all the scalesets is the right thing to do. Wouldn't that have an impact on customer workloads? It should be possible to restore clusters seemlessly. I would expect the cloud controller manager should be able to reconcile the current VMs with the node objects and delete stray objects, and kubelets with no node objects are able to recreate theirs.